bootstrap/pkg/kfupgrade/kfupgrade.go

```diff
@@ -75,12 +75,12 @@ func createNewKfApp(baseConfig string, version string, oldKfDef *kfdefsv3.KfDef)
 	newKfDef.Spec.Version = version
 
 	// Make sure the new KfApp is created.
-	_, err = coordinator.CreateKfAppCfgFile(newKfDef)
+	configFile, err := coordinator.CreateKfAppCfgFile(newKfDef)
```

Comment:
I don't think you want to call createKfAppCfgFile. CreateKfAppCfgFile should be an internal method used by NewKfApp which determines what to do when the argument to NewKfApp is a remote URI.

The KfUpgrade should determine the appropriate directory to store KfApp and just write the file to that directory.
i.e. by calling converters.WriteConfigFile
https://github.com/kubeflow/kubeflow/blob/9f5ef66ce3d12ff2b603d07330a16661caa0b286/bootstrap/pkg/apis/apps/configconverters/converters.go#L139